### PR TITLE
Update models.py and admin.py

### PR DESCRIPTION
**Pull request by Bera0422**
Models are created in models.py and registered in admin.py. Since Django automatically creates an autoincrement, primary key id field when not explicitly specified, I did not add a seperate id field. However I left in comments if they are preferred. Also fields SHOULD be reviewed carefully since there may be missed ones.

### CODE REVIEW

**Comment by hakanaktas0** [comment]
It may be different when we try to interact with the database by coding, but from what I have tried using the admin panel of Django, none of the fields in any of the tables can be empty(null). This may be problematic especially for papers without abstracts, authors or a specified year. Can you add this functionality to some of the fields if possible? 
Nice work btw, works really good.

File: practice_app/api/models.py [hunk 1]
```diff
@@ -1,3 +1,57 @@
...
+    owner = models.ManyToManyField(User, related_name='owners')
+    saver = models.ManyToManyField(User, related_name='savers', blank=True)
+
+class FollowRequest(models.Model):
+    sender = models.ForeignKey(User, on_delete=models.CASCADE, related_name='sender')
+    receiver = models.ForeignKey(User, on_delete=models.CASCADE, related_name='receiver')
+    status = models.CharField(max_length=50)
+
+class Follower(models.Model):
+    user = models.ForeignKey(User, on_delete=models.CASCADE)
```
Comment (hakanaktas0) [inline practice_app/api/models.py:55]: In this model, user is the one followed by the follower(s) and user follows followed(s). Right?

File: practice_app/api/models.py [hunk 2]
```diff
@@ -1,3 +1,57 @@
...
+class UserInterest(models.Model):
+    # interest_id = models.AutoField(primary_key=True)
+    user = models.ForeignKey(User, on_delete=models.CASCADE)
+    interest = models.CharField(max_length=50, blank=True)
+
+class PaperList(models.Model):
+    #list_id = models.AutoField(primary_key=True)
+    list_title = models.CharField(max_length=50, default='Paper List')
+    paper = models.ManyToManyField(Paper)
+    owner = models.ManyToManyField(User, related_name='owners')
```
Comment (hakanaktas0) [inline practice_app/api/models.py:46]: I think a paperlist should have only one owner.

File: practice_app/api/models.py [hunk 3]
```diff
@@ -1,3 +1,57 @@
...
+
+class UserInterest(models.Model):
+    # interest_id = models.AutoField(primary_key=True)
+    user = models.ForeignKey(User, on_delete=models.CASCADE)
+    interest = models.CharField(max_length=50, blank=True)
+
+class PaperList(models.Model):
+    #list_id = models.AutoField(primary_key=True)
+    list_title = models.CharField(max_length=50, default='Paper List')
+    paper = models.ManyToManyField(Paper)
```
Comment (hakanaktas0) [inline practice_app/api/models.py:45]: I think paper field can also be blank upon creation.

File: practice_app/api/models.py (hunk 2 above)
Comment (Bera0422) [reply practice_app/api/models.py]: Yes, I noticed and updated that.

File: practice_app/api/models.py (hunk 3 above)
Comment (Bera0422) [reply practice_app/api/models.py]: It makes sense, updated.

File: practice_app/api/models.py [hunk 4]
```diff
@@ -1,3 +1,57 @@
...
+
+class Paper(models.Model):
+    paper_id = models.AutoField(primary_key=True)
+    third_party_id = models.CharField(max_length=100)
+    source = models.CharField(max_length=50)
+    abstract = models.TextField(max_length=5000, blank=True, null=True)
+    year = models.IntegerField(blank=True, null=True)
+    url = models.URLField()
+    authors = models.TextField(max_length=500, blank=True, null=True)
+    title = models.CharField(max_length=100)
```
Comment (uyduranomer) [inline practice_app/api/models.py:16]: I think title max length should be more

File: practice_app/api/models.py [hunk 5]
```diff
@@ -1,3 +1,57 @@
 from django.db import models
+from django.contrib.auth.models import User
+
+import json
 
 # Create your models here.
+
+class Paper(models.Model):
+    paper_id = models.AutoField(primary_key=True)
```
Comment (uyduranomer) [inline practice_app/api/models.py:9]: As I remember, the source and the third_party_id were chosen as composit key in the meetings.
paper_id is not a problem, however, the combination of the third_party_id and source must be unique nevertheless.

File: practice_app/api/models.py [hunk 6]
```diff
@@ -1,3 +1,57 @@
...
+    interest = models.CharField(max_length=50, blank=True, null=True)
+
+class PaperList(models.Model):
+    #list_id = models.AutoField(primary_key=True)
+    list_title = models.CharField(max_length=50, default='Paper List')
+    paper = models.ManyToManyField(Paper, blank=True, null=True)
+    owner = models.ForeignKey(User, related_name='owner', on_delete=models.CASCADE)
+    saver = models.ManyToManyField(User, related_name='savers', blank=True, null=True)
+
+class FollowRequest(models.Model):
```
Comment (uyduranomer) [inline practice_app/api/models.py:49]: Can a user send a follow request to a user more than once ?

File: practice_app/api/models.py [hunk 7]
```diff
@@ -1,3 +1,57 @@
...
+    def get_authors(self):
+        return json.loads(self.authors) if self.authors else []
+
+class Comment(models.Model):
+    # comment_id = models.AutoField(primary_key=True)
+    user = models.ForeignKey(User, on_delete=models.CASCADE)
+    paper =  models.ForeignKey(Paper, on_delete=models.CASCADE)
+    text = models.TextField(max_length=500)
+
+class Like(models.Model):
```
Comment (uyduranomer) [inline practice_app/api/models.py:31]: Shouldn't combination of the user and paper be unique? I think composit keys should be used here.

**Review by uyduranomer — CHANGES_REQUESTED** [review]
I don't know if Django supports composit keys but I think we should use it when necessary. 
If it doesn't support it, then itshould be handled in the application level.

On the other hand, the models looks very nice. Good job!

File: practice_app/api/models.py (hunk 6 above)
Comment (Bera0422) [reply practice_app/api/models.py:49]: No, the status field in FollowRequest model will be used to reflect the current status of the request. Also I think this case can be handled in the implementation of the relevant POST method using the status field.

File: practice_app/api/models.py (hunk 4 above)
Comment (Bera0422) [reply practice_app/api/models.py]: I updated to 300.

File: practice_app/api/models.py (hunk 5 above)
Comment (hakanaktas0) [reply practice_app/api/models.py:9]: We decided to have a single paper_id in the meeting because it causes multiple foreign keys in all other tables.

Comment (Bera0422) [reply practice_app/api/models.py:9]: Yeah but I can still add a unique constraint for the third_party_id and source keeping the paper_id field. Other tables can access with paper_id.

File: practice_app/api/models.py (hunk 7 above)
Comment (Bera0422) [reply practice_app/api/models.py:35]: You're right. I added a unique constraint for those keys.

File: practice_app/api/models.py (hunk 5 above)
Comment (Bera0422) [reply practice_app/api/models.py:9]: I added a unique constraint for third_party_id and source keys.

File: practice_app/api/models.py (hunk 6 above)
Comment (uyduranomer) [reply practice_app/api/models.py:54]: Then, the follower and the receiver should not be composit key?

Comment (Bera0422) [reply practice_app/api/models.py:54]: You are right about the uniqueness of the sender and receiver fields. I will add uniqueness constraint for them as well.

**Comment by Bera0422** [comment]
> I think the follow request uniqueness should be handled in database level too since it might not be considered or misused in the application level.
> 
> The rest is fine with the latest changes I think. Thank you!

I added uniqueness constraint for the "receiver" and "sender" fields. Thanks for the feedback!

**Review by hakanaktas0 — APPROVED** [review]
All looks good.

**Review by uyduranomer — APPROVED** [review]
All looks good, good job my friend 👍

**Pull request merged**